internal/manager/run.go

```diff
@@ -312,3 +319,61 @@ func readyzHandler(mgr manager.Manager, dataplaneSynchronizer IsReady) func(*htt
 		return nil
 	}
 }
+
+type healthCheckHandler struct {
+	lock         sync.RWMutex
+	healthzCheck healthz.Checker
+	readyzCheck  healthz.Checker
+}
+
+func (s *healthCheckHandler) getHealthzCheck() healthz.Checker {
+	s.lock.RLock()
+	defer s.lock.RUnlock()
+
+	return s.healthzCheck
+}
+
+func (s *healthCheckHandler) getReadyzCheck() healthz.Checker {
+	s.lock.RLock()
+	defer s.lock.RUnlock()
+
+	return s.readyzCheck
+}
+
+func (s *healthCheckHandler) setHealthzCheck(checker healthz.Checker) {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	s.healthzCheck = checker
+}
+
+func (s *healthCheckHandler) setReadyzCheck(checker healthz.Checker) {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	s.readyzCheck = checker
+}
+
+func (s *healthCheckHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
+	var check healthz.Checker
+	switch req.URL.Path {
+	case "/healthz":
+		check = s.getHealthzCheck()
+	case "/readyz":
+		check = s.getReadyzCheck()
+	}
+
+	if check == nil {
+		rw.WriteHeader(http.StatusNotFound)
+		_, _ = rw.Write([]byte("not found"))
+		return
+	}
+
+	err := check(req)
+	if err != nil {
+		rw.WriteHeader(http.StatusInternalServerError)
+		_, _ = rw.Write([]byte(err.Error()))
+		return
+	}
+	_, _ = rw.Write([]byte(""))
```

Comment:
I don't see a reason to write an empty body. Much more descriptive would be to explicitly set the status code (200?)

Reply:
returned `ok` in the body to be consistent with the original one.